// react-use-id-hook/dist/react-use-id-hook.cjs.development.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var PropTypes = _interopDefault(require('prop-types'));

var Id =
/*#__PURE__*/
React.createContext(function () {
  throw new TypeError('Please wrap your application with IdProvider');
});

var useIdGetter = function useIdGetter(prefix) {
  if (prefix === void 0) {
    prefix = 'id';
  }

  var ref = React.useRef();

  if (!ref.current) {
    var me = {
      id: 0,
      get: function get() {
        return prefix + "-" + me.id++;
      }
    };
    ref.current = me;
  }

  return ref.current.get;
};

var IdProvider = function IdProvider(_ref) {
  var children = _ref.children;
  var get = useIdGetter();
  return React__default.createElement(Id.Provider, {
    value: get
  }, children);
};
IdProvider.propTypes = {
  children: PropTypes.node.isRequired
};
var useId = function useId() {
  var getter = React.useContext(Id);
  var ref = React.useRef();
  if (!ref.current) ref.current = getter();
  return ref.current;
};
var useGetId = function useGetId() {
  var getter = React.useContext(Id);
  var base = React.useRef();
  if (!base.current) base.current = getter();
  return useIdGetter(base.current);
};

exports.IdProvider = IdProvider;
exports.useGetId = useGetId;
exports.useId = useId;
